### ISSUE TITLE
Cursor jumps to component body instead of definition on first jump

### ISSUE THREAD

**Issue by gabrielvincent** [catgoose/templ-goto-definition#1]
The very first jump lands on the component body rather than on its definition, but subsequent jumps work as intended. See how when I go to definition from a `@components.Layout` instance, it first lands on the `<body>` tag the first time.

https://github.com/user-attachments/assets/6cc0b9a4-b52b-4e2e-98b6-9be70071b68b

**Comment by catgoose** [comment]
Could you provide a repo I could pull down to test against?

Thanks.

**Comment by catgoose** [comment]
Oh I see. I only tested this using go files not from templ. I'll look at this later today.

**Comment by gabrielvincent** [comment]
> Could you provide a repo I could pull down to test against?
> 
> Thanks.

Of course, here you go: https://github.com/gabrielvincent/templ-goto-definition-initial-gtd-bug

How to reproduce the error:
1. vim ./path/to/templ-goto-definition-initial-gtd-bug/.
2. Open `main.templ`.
3. Place the cursor over `Layout()`.
4. Hit `gd` keybinding.
5. See the error:
```
   Error  19:11:32 msg_show.lua_error Error executing vim.schedule lua callback: ...ar/neovim/0.10.1/share/nvim/runtime/lua/vim/lsp/util.lua:1914: index out of range
stack traceback:
	[C]: in function '_str_byteindex_enc'
	...ar/neovim/0.10.1/share/nvim/runtime/lua/vim/lsp/util.lua:1914: in function 'locations_to_items'
	...nvim/lazy/telescope.nvim/lua/telescope/builtin/__lsp.lua:197: in function 'handler'
	.../neovim/0.10.1/share/nvim/runtime/lua/vim/lsp/client.lua:687: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

How to get gtd action to work from the start:
1. vim ./path/to/templ-goto-definition-initial-gtd-bug/.
2. Open `layouts/layouts.templ`
2. Open `main.templ`.
3. Place the cursor over `Layout()`.
4. Hit `gd` keybinding.
5. Land on the definition correctly.

The problem only occurs when the destination buffer hasn't been opened yet.

**Comment by catgoose** [comment]
This looks exactly like https://github.com/a-h/templ/issues/893 which supposedly was fixed in the last release of templ, but I am getting the same error you are in .10.1 and nightly.

This is an issue with templ LSP it seems so I will open an issue with templ

**Comment by catgoose** [comment]
This should be fixed when templ makes a new release.  See: https://github.com/a-h/templ/issues/911

**Comment by catgoose** [comment]
@gabrielvincent can you see if this is working with latest templ version?
